docs/posts/flink-unified-stream-batch/index.qmd

```diff
@@ -0,0 +1,414 @@
+        "payAmount": dt.float64,
+        "payPlatform": dt.int32,
+        "provinceId": dt.int32,
+    }
+)
+
+env_settings = EnvironmentSettings.in_streaming_mode()
+table_env = TableEnvironment.create(env_settings)
+
+table_config = table_env.get_config()
+config = Configuration()
+config.set_string("parallelism.default", "1")
+table_config.add_configuration(config)
+
+connection = ibis.flink.connect(table_env)
+
+# add the JAR downloaded above
+connection.raw_sql("ADD JAR 'flink-sql-connector-kafka-1.17.1.jar'")
+
+source_configs = {
+    "connector": "kafka",
+    "topic": "payment_msg",
+    "properties.bootstrap.servers": "localhost:9092",
+    "properties.group.id": "test_3",
+    "scan.startup.mode": "earliest-offset",
+    "format": "json",
+}
+
+connection.create_table(
+    "payments",
+    schema=source_schema,
+    tbl_properties=source_configs,
+    watermark=ibis.watermark(
+        time_col="createTime", allowed_delay=ibis.interval(seconds=15)
+    ),
+)
+```
+
+How would we write this in Flink SQL? Ibis makes this extremely easy by
+exposing a `compile()` API:
+
+```{python}
+sql = connection.compile(unbound_agged)
+print(sql)
+```
+
+Before we can execute this query, we need to first define a data sink where the
+results can be written:
+
+```{python}
+sink_schema = sch.Schema(
+    {
+        "province_id": dt.int32,
+        "pay_amount": dt.float64,
+    }
+)
+
+kafka_sink_configs = {
+    "connector": "kafka",
+    "topic": "sink",
+    "properties.bootstrap.servers": "localhost:9092",
+    "format": "json",
+}
+
+connection.create_table(
+    "kafka_sink", schema=sink_schema, tbl_properties=kafka_sink_configs
+)
+```
+
+Now, let’s write the results into this sink. Note that we can directly reuse
+the transformation logic that we wrote above for the local execution backend!!
+
+```{python}
+connection.insert("kafka_sink", unbound_agged)
+```
+
+::: {.callout-tip}
+You can examine the results either using the Kafka console consumer CLI or the
+`kafka-python` library.
+:::
+
+How easy was it to define both batch and streaming workloads using Ibis?
+Without Ibis, you would have needed to write a `pandas`/DuckDB workload and
+then convert it into Flink SQL manually.
+
+## Concluding thoughts
+
+With the introduction of the first streaming backends, Ibis is now both a batch
+and a streaming Python DataFrame API and we’re excited about what’s to come
+next. We hope that Ibis can close the gap between batch and streaming in such a
+way that we no longer talk about the two separately, but, rather, as two parts
+of the same paradigm. Streaming naturally lends itself to batch: batch is
+technically just a special case of streaming, where the unbounded data flow
+stops at some point.
+
+Of course, this is only the beginning. There are still technical challenges to
+be solved (e.g., backfill, window computations over large windows, GPU
+acceleration), and we hope we'll have more exciting updates to share with the
```

Comment:
recommend removing "hope" here, was used in the last paragraph, and we'll definitely have more exciting updates to share!